tests/integration/test_manifests.py

```diff
@@ -0,0 +1,187 @@
+    # save the temp_model
+    temp_model = juju.model
+    # Switch to the k8s model
+    juju.model = microk8s_model
+    # Install resource dispatcher and its dependencies
+    juju.deploy(METACONTROLLER_CHARM, trust=True)
+    juju.deploy(ADMISSION_WEBHOOK_CHARM, trust=True)
+    juju.deploy(
+        RESOURCE_DISPATCHER,
+        trust=True,
+        channel=RESOURCE_DISPATCHER_CHANNEL,
+        app=RESOURCE_DISPATCHER_APP_NAME,
+    )
+
+    # Wait for charms to be active
+    status = juju.wait(
+        lambda status: jubilant.all_active(status) and jubilant.all_agents_idle(status), delay=5
+    )
+
+    logger.info(status)
+
+    # Offer interfaces
+    logger.info("Offering resource dispatcher interfaces")
+    juju.offer(
+        f"{microk8s_model}.{RESOURCE_DISPATCHER_APP_NAME}",
+        name="secrets-dispatcher",
+        endpoint="secrets",
+    )
+    juju.offer(
+        f"{microk8s_model}.{RESOURCE_DISPATCHER_APP_NAME}",
+        name="poddefaults-dispatcher",
+        endpoint="pod-defaults",
+    )
+
+    # Switch to the lxd model
+    logger.info("Switching to kubeflow-integrator model")
+    juju.model = temp_model
+
+    juju.deploy(
+        kubeflow_integrator,
+        app=KUBEFLOW_INTEGRATOR_APP_NAME,
+        config={"profile": "profile-name"},
+    )
+
+    # Consume offers
+    logger.info("Consuming offers")
+    juju.consume(f"{microk8s_model}.secrets-dispatcher")
+    juju.consume(f"{microk8s_model}.poddefaults-dispatcher")
+
+    logger.info("Waiting for all units to become active")
+    juju.wait(
+        lambda status: jubilant.all_active(status) and jubilant.all_agents_idle(status), delay=5
+    )
+
+    # Integrate kubeflow-integrator with resource-dispatcher
+    logger.info("Integrate kubeflow-integrator with resource-dispatcher")
+    juju.integrate(f"{KUBEFLOW_INTEGRATOR_APP_NAME}:secrets", "secrets-dispatcher")
+    juju.integrate(f"{KUBEFLOW_INTEGRATOR_APP_NAME}:pod-defaults", "poddefaults-dispatcher")
+
+    # Wait for the relatio data to be created
+
+    # Switch to k8s model
+    juju.model = microk8s_model
+
+    logger.info("Checking secrets and pod defaults relation data")
+
+    for attempt in Retrying(stop=stop_after_attempt(5), wait=wait_fixed(3), reraise=True):
+        with attempt:
+            # Check application data
+            secrets_rel_data = list(
+                get_application_data(juju, RESOURCE_DISPATCHER_APP_NAME, "secrets").items()
+            )[0]
+            assert secrets_rel_data[1] == {}
+            poddefaults_rel_data = list(
+                get_application_data(juju, RESOURCE_DISPATCHER_APP_NAME, "pod-defaults").items()
+            )[0]
+            assert poddefaults_rel_data[1] == {}
+
+    juju.model = temp_model
+
+
+def test_manifests_generation_with_opensearch(juju: jubilant.Juju, microk8s_model: str):
+    """Deploy opensearch, integrate it with kubeflow-integrator and make sure that manifests are generated in the relation data."""
```

Comment:
I would have expected the linting rules to complain about the length of these docstrings (which iirc should not exceed 100 chars)